send_test.go

```diff
@@ -229,3 +229,47 @@ func TestSendAtSingleMessage(t *testing.T) {
 	assert.Equal(messageID, result.Messages[0].ID)
 	assert.Equal(messageURI, result.Messages[0].URI)
 }
+
+func TestOriginatorValidation(t *testing.T) {
```

Comment:
Normally we avoid implementing validation in the SDKs and leave it up to the API. This is to avoid duplicating behaviour and increasing maintenance burden of change. If you strip the validation I think this PR is good to go.